### DISCUSSION TITLE
Does htaccess work on GitHub Pages?

### DISCUSSION THREAD

**Discussion by blacbrue** [community/community#23723] [Other Feature Feedback, Questions, & Ideas]
<p>So I wanted to take out the html extension from the url using htaccess. is there any possible way to do it?</p>

**Comment by kingthorin** [comment]
<p>If you’re using Jekyll you can access the generated content without html extension.</p>

**Comment by blacbrue** [comment]
<p>So .htaccess doesn’t work on GitHub Pages?</p>

**Comment by kingthorin** [comment]
<p>No idea, but GH pages do specifically support Jekyll. Jekyll’s rendered HTML can be referenced with or without file extension.</p>

**Comment by otter-computer** [comment, marked as answer]
<p>GitHub Pages doesn’t support <code>.htaccess</code>, no. Pages does support visiting pages without the <code>.html</code> extension and it will load correctly, but it won’t automatically redirect from the <code>.html</code> version to the URL without it.</p>

**Reply by kundalkr** [reply]
what i do for 301 redirection on website

**Comment by blacbrue** [comment]
<p>Okay. Thanks for your response</p>

**Comment by 3raxton** [comment]
Do we have any updates on this as of 2022?

**Comment by ethanlevy007** [comment]
I hope this will work in the future in GitHub. As[,](https://thedaomakers.com/) it is looking quite difficult but they have to consider this.

**Comment by pohlt** [comment]
Don't hold your breath for this feature as it is already available for GH Enterprise. I guess this feature will stay behind the paywall for the foreseeable future.

No hard feelings here. Somehow they have to make money.

**Comment by tylerjmorg** [comment]
It looks like `.htaccess` does work with `ErrorDocument 404` commands on GitHub Pages. But I have not been able to complete anything else using the file.

**Comment by Mihailoff** [comment]
https://github.com/server-info/.htaccess (half-serious, for the overall awareness)
